e2e/testcases/invalid_git_branch_test.go

```diff
@@ -29,6 +29,18 @@ import (
 
 func TestInvalidRootSyncBranchStatus(t *testing.T) {
 	nt := nomostest.New(t, nomostesting.SyncSource, ntopts.SkipMonoRepo)
+	nt.T.Cleanup(func() {
+		// Cleanup to Reset RootSync back to main.
+		nomostest.SetGitBranch(nt, configsync.RootSyncName, nomostest.MainBranch)
+		nt.WaitForRepoSyncs()
+
+		err := nt.ValidateMetrics(nomostest.SyncMetricsToLatestCommit(nt), func() error {
```

Comment:
Do we need to validate the metrics in the cleanup? Ideally the metrics would reflect the synced state, right?

Reply:
I think it is okay either way. No strong objection.

Reply:
Prior to this PR, the test body was:
1. cause problem
2. verify problem is reflected in sync status
3. verify problem is reflected in metrics
4. fix problem
5. verify fix is reflected in sync status
6. verify fix is reflected in sync metrics

This is a common pattern. Should steps 4-6 always be in Cleanup? Or should all the steps be in the test body and the cleanup be just step 4?

Reply:
Correction: the test is missing step 3. It should probably be added to validate the source error.

Reply:
I'd suggest having a combination of both. We can:
- Fix problem and make assertions in test body
- Reset branch in cleanup, without assertions

Reply:
@nan-yu @karlkfi Can we punt the test refactor into a separate PR where we can discuss the best pattern to use for test assertion and celanup? I'll add another commit to remove the cleanup changes made in `TestInvalidRootSyncBranchStatus` so that we can focus on fixing the broken `TestSyncFailureAfterSuccessfulSyncs` in the PR only.

Reply:
I think we can maintain the test body with the 6 steps as Karl commented, but i'm just thinking if there is a need for an explicit cleanup?
The test doesn't switch to a different branch in the repo and no resources are added in the test. Even if it fails before step 4, the generic reset function will re-initialize the git repo.